src/rpc/misc.cpp

```diff
@@ -484,24 +485,24 @@ UniValue setmocktime(const UniValue& params, bool fHelp)
     return NullUniValue;
 }
 
-bool getAddressFromIndex(const int &type, const uint160 &hash, std::string &address)
+bool getAddressFromIndex(const AddressType &type, const uint160 &hash, std::string &address)
 {
-    if (type == 2) {
+    if (type == AddressType::SCRIPT) {
         address = CBitcoinAddress(CScriptID(hash)).ToString();
-    } else if (type == 1) {
+    } else if (type == AddressType::PUBKEY) {
         address = CBitcoinAddress(CKeyID(hash)).ToString();
     } else {
         return false;
     }
     return true;
```

Comment:
When dissecting cases over an enum prefer the use of `switch` 

```suggestion
    switch (type) {
    case AddressType::SCRIPT:
        address = CBitcoinAddress(CScriptID(hash)).ToString();
        return true;
    case AddressType::PUBKEY:
        address = CBitcoinAddress(CKeyID(hash)).ToString();
        return true;
    default:
        return false;
    }
```

Reply:
Generally I'd agree with your opinion but I am not so sure in case of a very short set valued enum (like in this case, just 3 possible values); can you please share your view on this?

Reply:
There are many reasons to prefer that:

- an `if ... else` uses multiple statements for  multiple choices while `switch` uses a single expression
- `if` can evaluate multiple conditions for **any** logical expression while `case` can compare **only** for equality of chars or integer values
- as the main goal (in this case) is traverse an enum, as the switch can't have two identical `case` values we're implicitly safe against possible errors like ` if (type == AddressType::SCRIPT) {...} else if (type == AddressType::SCRIPT) { ...}` which builds correctly but obviously is not what we want to see.
- efficiency : compiler optimizes (at any O level) `switch` over `if ... else` (see https://godbolt.org/z/8r4shrKPG)

Reply:
Done